source/eitri/eitri.py

```diff
@@ -44,43 +52,121 @@
-    :param output_file: the output file path
-    :param ed_component_path: the folder path containing all "*2ed" and "ed2nav" binaries
-    :param cities_exec_path: the path to the binary "cities"
-    :param import_cities: the path to cities file (ex: france_boundaries.osm.pbf) to load
+    :return: the path to the directory with all data. If several datasets ("*.osm", "*.gtfs", ...) are available, they need to be in separate directories
+    :return: the output file path
+    :return: the path of the folder containing all "*2ed" and "ed2nav" binaries
+    :return: the path to the binary "cities"
+    :return: the path to cities file (ex: france_boundaries.osm.pbf) to load
     """
+    parser = argparse.ArgumentParser(
+        prog='eitri',
+        description='Eitri uses a set of binaries to extract data, load it into a database and extract it to a single file. '
+        'This software is not distributed and only used by Navitia.',
+        epilog='For more informations, see www.navitia.io',
+    )
+    parser.add_argument(
+        '-d',
+        '--data-folder',
+        required=True,
+        type=is_folder,
+        help='the directory path with all data. If several datasets ("*.osm", "*.gtfs", ...) are available, they need to be in separate directories',
+    )
+    parser.add_argument(
+        '-e',
+        '--ed-folder',
+        required=True,
+        type=is_folder,
+        help='the directory path containing all "*2ed" and "ed2nav" binaries',
+    )
+    parser.add_argument(
+        '-o',
+        '--output-file',
+        type=is_file_exist,
+        default='./data.nav.lz4',
+        help="the output file path (default: './data.nav.lz4')",
+    )
+    parser.add_argument(
+        '-i', '--cities-file', type=is_file, help='the path to the "cities" file to load  (usually a *.osm.pbf)'
+    )
+    parser.add_argument(
+        '-f', '--cities_folder', type=is_folder, help='the path to the folder containing the "cities" binary'
+    )
+    parser.add_argument(
+        '-c', '--color', type=is_enabled, default=True, help='enable or disable colored logs (default: True)'
+    )
+    return parser.parse_args()
 
-    # there is some problems with environment variables and cmake, so all args can be given through cli
-    for p in add_pythonpath:
-        sys.path.append(p)
 
-    from ed_handler import generate_nav
-    from docker_wrapper import PostgresDocker
+def eitri():
+    # type: () -> None
+    """
+    Generate a "data.nav.lz4" file.
+    """
+    args = get_params()
+    if args.color:
+        log.enable_color(True)
+    else:
+        log.enable_color(False)
 
-    with closing(PostgresDocker()) as docker_ed, closing(PostgresDocker()) as docker_cities:
-        generate_nav(
-            data_dir,
+    logger.info("starting Eitri")
+    with contextlib.closing(docker_wrapper.PostgresDocker()) as docker_ed, contextlib.closing(
```

Comment:
I think it's better to show the package name instead of just its functions, it improves readability even if in that case I agree the package name is long. What do you think about "ctx.closing()" ?

Reply:
don't worry, I can live with it ;)